main.js

```diff
@@ -52,10 +52,39 @@ function RoundToHundredths(x) {
   return Math.floor(x * 100 + 0.5) / 100;
 }
 
-function HandleNode(svgNode, scaleX, scaleY, translateX, translateY) {
+// |fillString| is expected to be "#RRGGBB" or "#RGB"].
+function PathColorFromFill(fillString) {
+  if(fillString.length === 4) {
+    // Color in form of #RGB so let's turn that to #RRGGBB.
+    fillString = `#${fillString[1]}${fillString[1]}${fillString[2]}${fillString[2]}${fillString[3]}${fillString[3]}`.toUpperCase();
+  }
+
+  const r = fillString.substr(1,2);
+  const g = fillString.substr(3,2);
+  const b = fillString.substr(5,2);
+  
+  return `PATH_COLOR_ARGB, 0xFF, 0x${r}, 0x${g}, 0x${b},\n`;
+}
+
+function HandleNode(svgNode, scaleX, scaleY, translateX, translateY, preserveFill) {
```

Comment:
(I don't know why github doesn't let me comment on unchanged lines)

This is also called from inside the 'g' case, shouldn't we pass preserveFill there?